react/setup-modal.jsx

```diff
@@ -69,9 +69,11 @@ let SetupModal = React.createClass({
     return asink(function *() {
       this.close()
       let dattcore = this.props.dattcore
-      yield dattcore.asyncSetUserName(newUserName)
       yield dattcore.asyncSetUserSetupFlag(true)
+      let userNameSynced = dattcore.asyncSetUserName(newUserName) // *intentionally* NOT yield-ing
       window.location.hash = '#/frontpage'
+      yield userNameSynced
+      dattcore.emit('refresh-content', null)
```

Comment:
And, on that note, if you find you need to emit events in the UI, this may be time to use Flux, which I believe was invented for that purpose: https://facebook.github.io/flux/docs/overview.html